src/controller/CommissioneeDeviceProxy.h

```diff
@@ -0,0 +1,392 @@
+
+    void Reset();
+
+    NodeId GetDeviceId() const override { return mDeviceId; }
+
+    bool MatchesSession(SessionHandle session) const { return mSecureSession.HasValue() && mSecureSession.Value() == session; }
+
+    chip::Optional<SessionHandle> GetSecureSession() const override { return mSecureSession; }
+
+    Messaging::ExchangeManager * GetExchangeManager() const override { return mExchangeMgr; }
+
+    void SetAddress(const Inet::IPAddress & deviceAddr) { mDeviceAddress.SetIPAddress(deviceAddr); }
+
+    PASESession & GetPairing() { return mPairing; }
+
+    uint8_t GetNextSequenceNumber() override { return mSequenceNumber++; };
+
+    CHIP_ERROR LoadSecureSessionParametersIfNeeded()
+    {
+        bool loadedSecureSession = false;
+        return LoadSecureSessionParametersIfNeeded(loadedSecureSession);
+    };
+
+    CHIP_ERROR SetCSRNonce(ByteSpan csrNonce)
+    {
+        VerifyOrReturnError(csrNonce.size() == sizeof(mCSRNonce), CHIP_ERROR_INVALID_ARGUMENT);
+        memcpy(mCSRNonce, csrNonce.data(), csrNonce.size());
+        return CHIP_NO_ERROR;
+    }
+
+    ByteSpan GetCSRNonce() const { return ByteSpan(mCSRNonce, sizeof(mCSRNonce)); }
+
+    CHIP_ERROR SetAttestationNonce(ByteSpan attestationNonce)
+    {
+        VerifyOrReturnError(attestationNonce.size() == sizeof(mAttestationNonce), CHIP_ERROR_INVALID_ARGUMENT);
+        memcpy(mAttestationNonce, attestationNonce.data(), attestationNonce.size());
+        return CHIP_NO_ERROR;
+    }
+
+    ByteSpan GetAttestationNonce() const { return ByteSpan(mAttestationNonce, sizeof(mAttestationNonce)); }
+
+    bool AreCredentialsAvailable() const { return (mDAC != nullptr && mDACLen != 0); }
+
+    ByteSpan GetDAC() const { return ByteSpan(mDAC, mDACLen); }
+    ByteSpan GetPAI() const { return ByteSpan(mPAI, mPAILen); }
+
+    CHIP_ERROR SetDAC(const ByteSpan & dac);
+    CHIP_ERROR SetPAI(const ByteSpan & pai);
+
+    MutableByteSpan GetMutableNOCCert() { return MutableByteSpan(mNOCCertBuffer, sizeof(mNOCCertBuffer)); }
+
+    CHIP_ERROR SetNOCCertBufferSize(size_t new_size);
+
+    ByteSpan GetNOCCert() const { return ByteSpan(mNOCCertBuffer, mNOCCertBufferSize); }
+
+    MutableByteSpan GetMutableICACert() { return MutableByteSpan(mICACertBuffer, sizeof(mICACertBuffer)); }
+
+    CHIP_ERROR SetICACertBufferSize(size_t new_size);
+
+    ByteSpan GetICACert() const { return ByteSpan(mICACertBuffer, mICACertBufferSize); }
+
+    Controller::DeviceControllerInteractionModelDelegate * GetInteractionModelDelegate() override { return mpIMDelegate; };
+
+private:
+    enum class ConnectionState
+    {
+        NotConnected,
+        Connecting,
+        SecureConnected,
+    };
+
+    enum class ResetTransport
+    {
+        kYes,
+        kNo,
+    };
+    /* Node ID assigned to the CHIP device */
+    NodeId mDeviceId;
+
+    /** Address used to communicate with the device.
+     */
+    Transport::PeerAddress mDeviceAddress = Transport::PeerAddress::UDP(Inet::IPAddress::Any);
```

Comment:
Might it make more sense to initialize with the default PeerAddress() constructor?  That has mTransportType(Type::kUndefined), which seems like it would make more sense since we don't know whether we will be using UDP or TCP.

Reply:
I'll check and update accordingly.